llvm/test/CodeGen/RISCV/GlobalISel/legalizer/rv32/legalize-vadd.mir

```diff
@@ -0,0 +1,9 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
```

Comment:
Maybe we should append vector add cases to legalize-add.mir file instead of creating a new one. It looks like that legalize-add.mir file is testing G_ADD leglization on different types.

Reply:
Also because `vadd` is not the GISel opcode we are legalize. We are legalizing a G_ADD opcode that operates on vector types.

Reply:
Sounds reasonable! And I tried to change it, and used `SelectionDAG` to run. But again, it fails at 
```
/llvm-project/llvm/include/llvm/CodeGen/MachineRegisterInfo.h:665: const TargetRegisterClass *llvm::MachineRegisterInfo::getRegClass(Register) const: Assertion `isa<const TargetRegisterClass *>(VRegInfo[Reg.id()].first) && "Register class not set, wrong accessor"' failed.
```

Nevertheless, as you comment before:
> I touched this function in one of my PRs: https://github.com/llvm/llvm-project/pull/70881. Maybe you can investigate why we're failing this assertion as a result of this test case? Were we supposed to assign a register a regclass? Maybe we need to do some work in RegBankSelection to avoid this failure?

> I am not sure why the failure happens with the options you give it. However, I will note that SDAG works on LLVM IR, not MIR so I don't think you'd really get the result you intended on getting anyway.

Let me just proceed with `-global-isel` and see if it will fail at the same spot. If it does, I will take a closer look at it; for now, I'll just ignore `SelectionDAG`'s error since it's an MIR test and that 
> I will note that SDAG works on LLVM IR, not MIR so I don't think you'd really get the result you intended on getting anyway

What do you think